torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -360,10 +364,12 @@ def wrapper(
         self, *args: Tuple[object], **kwargs: Dict[str, Any]  # type: ignore[misc]
     ) -> None:
         # if enough GPU we can use GPU, otherwise we fallback to CPU
-        if not torch.cuda.is_available() or torch.cuda.device_count() < self.world_size:
-            self.device_type = "cpu"
+        if torch.cuda.is_available() and torch.cuda.device_count() >= self.world_size:
+            self.device_type = "cuda"
+        elif TEST_HPU and torch.hpu.device_count() >= self.world_size:
+            self.device_type = "hpu"
         else:
-            self.device_type = DEVICE_TYPE
+            self.device_type = "cpu"
```

Comment:
Why was it changed?

Reply:
DEVICE_TYPE was actually redundant, device type is updated based on the device , however here we update the device for pg create to the device only if number of devices  is >= world_size, which is again a hard coded value of 4